libarchive/archive_entry.h

```diff
@@ -115,6 +118,14 @@ typedef ssize_t la_ssize_t;
 #define __LA_DEV_T la_int64_t
 #endif
 
+#if ARCHIVE_VERSION_NUMBER < 4000000
+/* Libarchive 3.x used signed int64 for inode numbers */
+#define __LA_INO_T la_int64_t
```

Comment:
Define a new `__LA_INO_T` for inode values; it will remain signed in libarchive 3 and switch to unsigned in libarchive 4.

Reply:
Should we also transition some other fields to unsigned in libarchive 4?